src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -68,7 +68,7 @@ public function createQueryBuilderByChannelAndTaxonSlug(ChannelInterface $channe
             ->andWhere('channel = :channel')
             ->andWhere('o.enabled = true')
             ->andWhere('channelPricing.channel = :channel')
-            ->groupBy('o.id')
+            ->groupBy('o.id, translation.id, productTaxon.position')
```

Comment:
That sorting crash tests because values tested are not the right ones.

Are you sure ? The original groupBy has his own "sorting". Usually, sorty is managed in orderBy no ?

Also, grouping on price may produces non unique results (one per channel)